allennlp/data/vocabulary.py

```diff
@@ -16,6 +16,8 @@
 from allennlp.common.params import Params
 from allennlp.common.checks import ConfigurationError
 from allennlp.common.file_utils import cached_path
+from allennlp.data import instance as adi  # pylint: disable=unused-import
```

Comment:
Can you not just change the position of the imports in `__init__` of `data` so you can remove this?

Reply:
I don't think so, the problem is that `Instance.index_fields` needs to know about `Vocabulary`, which means that we get a circular reference if we import `Instance` here directly.

this is the solution I got from https://www.python.org/dev/peps/pep-0484/#forward-references